api/rosetta/data_block_integration_test.go

```diff
@@ -0,0 +1,322 @@
+// Copyright 2021 Alvalor S.A.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+// +build integration
+
+package rosetta_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+	"testing"
+
+	"github.com/labstack/echo/v4"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/optakt/flow-dps/api/rosetta"
+	"github.com/optakt/flow-dps/models/dps"
+	"github.com/optakt/flow-dps/rosetta/identifier"
+	"github.com/optakt/flow-dps/rosetta/object"
+)
+
+type transactionValidationFn func(*object.Transaction)
+
+func TestGetBlock(t *testing.T) {
+
+	db := setupDB(t)
+	api := setupAPI(t, db)
+
+	tests := []struct {
+		name string
+
+		request rosetta.BlockRequest
+
+		wantStatusCode       int
+		wantTimestamp        int64
+		wantParentHash       string
+		wantHandlerErr       assert.ErrorAssertionFunc
+		transactionValidator transactionValidationFn
+	}{
+		{
+			// TODO: this is a natural boundary element, but the parent block we will receive will be a bit weird (parent ID is uint64(-1));
+			// since the recent rosetta explicitly states that block with height 0 is not supported, probably should be removed
+			name:           "first block",
```

Comment:
Since #159 explicitly states that request for block with height 0 is not supported, this should probably be removed